### DISCUSSION TITLE
How do I load_asset_value from a job executed in process with mem_io_manager?

### DISCUSSION THREAD

**Discussion by pedrovgp** [dagster-io/dagster#10781] [Q&A] [answered]
Also [in stackoverflow](https://stackoverflow.com/questions/74615651/in-dagster-how-do-i-load-asset-value-from-a-job-executed-in-process-with-mem-io).

For this question, consider I have a repository with one asset:
```python
@asset
def my_int():
    return 1

@repository
def my_repo():
    return [my_int]
```
I want to execute it in process (with mem_io_manager), but I would like to retrieve the value returned by my_int from memory later. I can do that with fs_io_manager, for example, using `my_repo.load_asset_value('my_int')`, after it ran. But the same method with mem_io_manager raises `dagster._core.errors.DagsterInvariantViolationError: Attempting to access step_key, but it was not provided when constructing the OutputContext`.

Ideally, I would execute it in process and tell the executor to return me one (or more) of the assets, something like:
```python
my_assets = my_repo.get_job('__ASSET_JOB').execute_in_process(return_assets=[my_int, ...])
```

**Comment by dpeng817** [comment, marked as answer]
you can do this with materialize_to_memory: 
```python
my_materialize_asset = materialize_to_memory([asset]).output_for_node(asset.node_def.name)
```

**Reply by pedrovgp** [reply]
Indeed, thanks, I later found the `output_for_node`. In the example above, it would work to do:
```python
result = my_repo.get_job('__ASSET_JOB').execute_in_process()
my_int = result.output_for_node("my_int")
```
